### ISSUE TITLE
False negative for ``implicit-str-concat`` when it happens on a single line inside a multiline string with ``check-str-concat-over-line-jumps=no``

### ISSUE THREAD

**Issue by Pierre-Sassoulas** [PyCQA/pylint#6444]
### Bug description

```python
# There's an internal juxtaposition on the second line in a valid multiline
print(
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit,"
    " sed do eiusmod tempor " "incididunt ut labore et dolore "  # [implicit-str-concat]
    "magna aliqua. Ut enim ad minim veniam, quis nostrud "
    "exercitation ullamco laboris nisi ut aliquip ex ea "
)
```

This is hard to fix because we use the tokenisation to detect this issue and the current tokenisation do not permit to handle it easily so it will have to be modified first.

### Configuration

```ini
[STRING]
check-str-concat-over-line-jumps=no
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)
```

### Expected behavior

``implicit-str-concat`` on the second line.

### Pylint version

```shell
pylint 2.14.0-dev0
astroid 2.11.3
Python 3.8.10 (default, Mar 15 2022, 12:22:08) 
[GCC 9.4.0]
```
